src/rest.js

```diff
@@ -57,11 +57,18 @@ function del(config, auth, className, objectId) {
     }
     return Promise.resolve({});
   }).then(() => {
+    if (!auth.isMaster) {
+      return auth.getUserRoles();
```

Comment:
we could add an auth.resolveACL() that would return a proper ACL query object given the current state of the auth.
That would remove the duplication of concatenating with userRoles at both places